sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereStatement.cs

```diff
@@ -31,13 +31,22 @@ public WhereLogic Where()
         /// <inheritdoc/>
         public override AdtQueryBuilder Build()
         {
-            throw new InvalidOperationException("Invalid query: Missing WHERE logic.");
+            return _parent;
         }
 
         /// <inheritdoc/>
         public override string GetQueryText()
         {
-            return QueryConstants.Where;
+            string whereLogicString = _upstreamWhereLogic.GetQueryText();
+
+            if (whereLogicString.Length > 0)
+            {
+                return $"{QueryConstants.Where} " + whereLogicString;
+            }
+            else
+            {
+                return whereLogicString;
```

Comment:
you don't need an else statement, just returning an empty string should suffice.

```
if ( .... ) {
    .
    . 
    .
    return .... ;
}

return string.Empty;
```